api/v1beta1/common.go

```diff
@@ -25,43 +25,105 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
-// NodeSection is a specification of the node attributes
+// AnsibleOpts defines a logical grouping of Ansible related configuration options.
+type AnsibleOpts struct {
+	// AnsibleUser SSH user for Ansible connection
+	// +kubebuilder:validation:Optional
+	AnsibleUser string `json:"ansibleUser,omitempty"`
+
+	// AnsibleHost SSH host for Ansible connection
+	// +kubebuilder:validation:Optional
+	AnsibleHost string `json:"ansibleHost,omitempty"`
+
+	// AnsiblePort SSH port for Ansible connection
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:number"}
+	AnsiblePort int `json:"ansiblePort,omitempty"`
+
+	// AnsibleVars for configuring ansible
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +kubebuilder:validation:Schemaless
+	AnsibleVars map[string]json.RawMessage `json:"ansibleVars,omitempty"`
+
+	// AnsibleSSHPrivateKeySecret Private SSH Key secret containing private SSH
+	// key for connecting to node. Must be of the form:
+	// Secret.data.ssh-privatekey: <base64 encoded private key contents>
+	// <https://kubernetes.io/docs/concepts/configuration/secret/#ssh-authentication-secrets>
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:io.kubernetes:Secret"}
+	AnsibleSSHPrivateKeySecret string `json:"ansibleSSHPrivateKeySecret,omitempty"`
```

Comment:
I don't see how this can be set per-node, as setting NodeSet.Spec.NodeTemplate.Nodes[XX].Ansible.AnsibleSSHPrivateKeySecret would have no effect. 

AnsibleSSHPrivateKeySecret probably just needs to be under NodeSection for the time being. Later on, if we want to make it settable per-node, we'd need to add functionality to check if it's set and add the secret to ExtraMounts, then set an Ansible var on the node in the inventory to use the right key path. But all that is best saved for a different PR, if we ever want to add it.